src/main/java/appeng/client/render/overlay/OverlayRenderType.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * This file is part of Applied Energistics 2.
+ * Copyright (c) 2021, TeamAppliedEnergistics, All rights reserved.
+ *
+ * Applied Energistics 2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Applied Energistics 2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.client.render.overlay;
+
+import java.util.OptionalDouble;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+
+import org.lwjgl.opengl.GL11;
+
+import net.minecraft.client.renderer.IRenderTypeBuffer;
+import net.minecraft.client.renderer.RenderType;
+import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
+import net.minecraft.client.renderer.vertex.VertexFormat;
+
+/**
+ * This is based on the area render of https://github.com/TeamPneumatic/pnc-repressurized/
+ */
+public class OverlayRenderType extends RenderType {
+
+    public OverlayRenderType(String p_i225992_1_, VertexFormat p_i225992_2_, int p_i225992_3_, int p_i225992_4_,
+            boolean p_i225992_5_, boolean p_i225992_6_, Runnable p_i225992_7_, Runnable p_i225992_8_) {
+        super(p_i225992_1_, p_i225992_2_, p_i225992_3_, p_i225992_4_, p_i225992_5_, p_i225992_6_, p_i225992_7_,
+                p_i225992_8_);
+    }
+
+    public static RenderType getBlockHilightFace() {
+        return makeType("block_hilight",
+                DefaultVertexFormats.POSITION_COLOR, GL11.GL_QUADS, 256,
```

Comment:
Please check what the 256 was for... Was this the max size of the buffer or was this just the size until it flushes?
Might hit this limit when showing very large area

(same below)

Reply:
No idea here. That particular source is also broken, so I cannot follow it at all.